src/bidiMapper/domains/input/ActionDispatcher.ts

```diff
@@ -69,14 +73,25 @@ async function getElementCenter(
   return {x: x as number, y: y as number};
 }
 
+async function isMacOS(context: BrowsingContextImpl) {
+  const {result} = await (
+    await context.getOrCreateSandbox(undefined)
+  ).callFunction(IS_MAC_DECL, {type: 'undefined'}, [], false, 'none', {});
+  assert(result.type !== 'exception');
+  assert(result.result.type === 'boolean');
+  return result.result.value;
+}
+
 export class ActionDispatcher {
   #tickStart = 0;
   #tickDuration = 0;
   #inputState: InputState;
   #context: BrowsingContextImpl;
+  #isMacOS: Promise<boolean>;
   constructor(inputState: InputState, context: BrowsingContextImpl) {
     this.#inputState = inputState;
     this.#context = context;
+    this.#isMacOS = isMacOS(this.#context);
```

Comment:
here an unhandled promise rejection might occur: please add a catch

Reply:
In what situation?

Reply:
when the async calls inside isMacOS throw (for example, target shutting down or any other reason really)